.github/workflows/ops.test.yml

```
@@ -0,0 +1,50 @@
+name: test
+
+on: [ push ]
+
+jobs:
+
+ test:
+  runs-on: ${{ matrix.os }}
+  strategy:
+   matrix:
+    os: [
+     ubuntu-latest,
+     macos-latest,
+     # @todo windows support
+     # windows-latest
+    ]
+    toolchain: [ stable, beta, nightly ]
+  steps:
+   - uses: actions/checkout@v2
+
+   - name: rust cache
+     uses: actions/cache@v2
+     env:
+      cache-name: cache-rust
+     with:
+      path: |
+        ~/.cargo/registry
+        ~/.cargo/git
+        target
+      key: ${{ matrix.os }}-cargo-${{ matrix.toolchain }}-${{ hashFiles('Cargo.toml') }}-${{ hashFiles('crates/**/Cargo.toml') }}
+      restore-keys: |
+       ${{ runner.os }}-cargo-${{ env.cache-name }}-
+       ${{ runner.os }}-cargo-
+       ${{ runner.os }}-
+
+   - uses: actions-rs/toolchain@v1
+     with:
+       toolchain: ${{ matrix.toolchain }}
+
+   # smoke test a release build
+   - uses: actions-rs/cargo@v1
+     with:
+       command: build
+       args: --release
+
+   # run all the tests
+   - uses: actions-rs/cargo@v1
+     with:
+       command: test
+       args: --release
```

Tests should not be run in release mode - when there are errors, especially flaky errors, we want to see the line numbers.

```suggestion
       args:
```